server/routerlicious/packages/services/src/mongodb.ts

```diff
@@ -106,6 +106,21 @@ export class MongoCollection<T> implements core.ICollection<T> {
             : { value, existing: false };
     }
 
+    public async findAndUpdate(query: any, value: T): Promise<{ value: T; existing: boolean; }> {
```

Comment:
The only difference between this and `findOrCreate` is the `upsert: true,` option. Instead of duplicating this method, would it be possible to make the existing implementation handle both cases?

Reply:
Yeah. I thought we can use it for both cases, however, the findOrCreate uses $setOnInsert.
Here is the definition:
https://www.mongodb.com/docs/manual/reference/operator/update/setOnInsert/

If we find an existing document, we won't update it by using $setOnInsert